src/prototype_structure.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+/**
+ * Provides structure memory.
+ */
+Object.defineProperty(Structure.prototype, "memory", {
+  get: function () {
+    if(Memory.structures === undefined) {
+      Memory.structures = {};
+    }
+    if(Memory.structures[this.id] === undefined) {
+      Memory.structures[this.id] = {};
+    }
+    return Memory.structures[this.id];
+  },
+  set: function(v) {
+    return _.set(Memory, 'structures.' + this.id, v);
+  },
+  configurable: true,
+  enumerable: false
+});
+
+Structure.prototype.getFutureEnergy = function() {
+  if (this.memory.futureEnergy === undefined) {
+    return 0;
+  }
+  return this.memory.futureEnergy;
+};
+
+Structure.prototype.changeFutureEnergy = function(amount) {
```

Comment:
+1, my idea is kind of half baked. It needs to be more robust and separate from the creep which can die or decide to do something else.